dask/bag/tests/test_bag.py

```diff
@@ -1581,3 +1581,25 @@ def test_dask_layers_to_delayed():
     assert i.dask.layers.keys() == {"delayed-" + name}
     assert i.dask.dependencies == {"delayed-" + name: set()}
     assert i.__dask_layers__() == ("delayed-" + name,)
+
+
+def test_to_dataframe_optimize_graph():
```

Comment:
You might consider structuring this test differently. Checking for specific key names is okay, but can feel a little brittle.

```python
from dask.utils_test import hlg_layer_topological

...
with dask.annotate(foo=True):
    y = x.map(...)
...
# just verify the `map`s aren't fused already
assert len(d.dask) > d.npartitions

d = y.to_dataframe()
# Graph has been fused, so there is one task per partition
assert len(d.dask) == d.npartitions

d2 = y.to_dataframe(optimize_graph=False)
# Graph hasn't been fused. It should contain all the original keys, plus a `map_partitions` step converting to DataFrames
assert d.dask.keys().issubset(d2.dask.keys())
assert len(d2.dask) == len(d.dask) + d.npartitions
# Annotations are still there
assert hlg_layer_topological(d2.dask, 1).annotations == {"foo": True}
```

Reply:
Can't get the `issubset` assertion to work. The problem is that the graph of `d` contains an operation with prefix `lambda-to-dataframe`, which is not present in the graph of `d2`.

Reply:
I bet I meant to say `y` instead of `d`, like `assert y.dask.keys().issubset(d2.dask.keys())`